common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleServiceMetrics.java

```diff
@@ -84,13 +83,13 @@ public static void collectMetric(
         .addGauge(new ShuffleServiceMetricsInfo(name + "_rateMean", "Mean rate of timer " + name),
           t.getMeanRate())
         .addGauge(
-          getShuffleServiceMetricsInfoForGenericValue(timingName, "max"), snapshot.getMax())
+          getShuffleServiceMetricsInfoForGenericValue(name, "max"), snapshot.getMax())
```

Comment:
I'm not familiar with the metrics part. Just wonder, will the naming change break the backward compatibility?

Reply:
It would, yes, but the `Snapshot` portion of `Timer` metrics were only exposed from `YarnShuffleServiceMetrics` starting after PR #32388 which hasn't been released yet (slated for 3.2.0). Part of why I'd like to get this PR in soon is to have it before 3.2.0 goes out, so that we can update these metric names w/o worrying about compatibility.

Reply:
sgtm